docs/security.md

```diff
@@ -0,0 +1,182 @@
+  further limit access to the secrets, which can then have a lower scope.
+
+- **Mask secrets** - Ensure secrets are never logged or exposed during workflow executions. GitHub automatically masks
+  secrets, however they should still never be printed or written to files.
+
+## 1.2. `GITHUB_TOKEN` permissions
+
+- **Use minimal permissions** - Grant the least amount of permissions required to perform a job. For example, jobs that
+  build and test a project rarely require any permissions other than `contents: read`.
+
+- **Always override the default permissions** - Workflows should always specify the permissions for a job or workflow.
+  The base permissions should always be `contents: read`.
+
+Use the `permissions:` key at the workflow or job level to explicitly declare the permissions needed. Avoid relying on
+the default GitHub token permissions.
+
+A full list of permissions can be found
+on [GitHub's documentation: Controlling permissions for GITHUB_TOKEN](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token).
+
+```yaml
+permissions:
+  contents: read
+  id-token: write # Only when using OIDC.
+```
+
+## 1.3. Actions
+
+- **Use trusted actions** - Prefer actions from trusted sources, such as official GitHub actions or those hosted in the
+  `hemilabs/actions` repository. Always review third-party (including GitHub) actions prior to using them.
+
+- **Check transitive dependencies** - Actions, especially composite actions, may depend on other actions. Such
+  dependencies should be reviewed (usually possible by viewing the `action.yml` file within the action repository) and
+  ensured that all transitive dependencies are pinned to commit hashes.
+
+- **Pin actions to commit hashes** - Third-party actions **must be pinned to commit hashes** to avoid unexpected
+  changes. Only commit hashes for released tags should be used, and a comment should be added to the end of the line to
+  document the release that is in use. **Never pin to `latest`, branches or tags.** The only exception to this rule is
+  actions within `hemilabs/actions`, which should be pinned to a `vX.X` or `vX.X.X` tag.
+
+```yaml
+- uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+```
+
+---
+
+## 2. Actions
+
+## 2.1. Secrets
+
+- **Consuming secrets** - If an action requires secrets, ensure they are passed securely through GitHub Actions secrets.
+- **Mask sensitive data** - Ensure sensitive data is always masked in logs to avoid accidental exposure.
+
+## 2.2. Design
+
+- **Avoid external dependencies unless necessary** - When creating a custom action, avoid unnecessary dependencies to
+  minimise possible attack surface.
+
+- **Pin dependencies** - For composite actions, all third-party actions **must be pinned to commit hashes**. Similarly,
+  all JavaScript dependencies should be pinned to avoid unexpected and unreviewed changes.
```

Comment:
Truted/very-well-known 3rd party actions could be pinned by tag.

Reply:
https://github.com/hemilabs/actions/pull/18#discussion_r2068774704

Reply:
Updated to add exclusions for actions maintained by trusted organisations.